ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/NullScanTaskDispatcher.java

```diff
@@ -159,21 +165,46 @@ private void processTableScans(MapWork work, Set<TableScanOperator> tableScans)
           } else {
             managedEmptyPathMap.put(path, false);
           }
+        } else {
+          numberOfUnmanagedPaths++;
         }
       }
     }
 
+    int fetcherPoolParallelism = Math.min(MAXIMUM_ASYNC_STATUS_FETCHER_COUNT, (int)Math.sqrt(numberOfUnmanagedPaths));
```

Comment:
Would be good to reuse HIVE_COMPUTE_SPLITS_NUM_THREADS instead of MAXIMUM_ASYNC_STATUS_FETCHER_COUNT. On need basis, it can be set to "1" to get back old behaviour.